CMakeLists.txt

```diff
@@ -0,0 +1,69 @@
+CMAKE_MINIMUM_REQUIRED(VERSION 3.0)
+
+option(HUNTER_ENABLED "Enable Hunter package manager" NO)
+include("cmake/Hunter/HunterGate.cmake")
+HunterGate(
+        URL "https://github.com/ruslo/hunter/archive/v0.19.167.tar.gz"
+        SHA1 "244a13718ea2e6c0c8998d15559c7e459f615708"
+)
+
+PROJECT(SDL_ttf VERSION 2.0.14)
+
+hunter_add_package(freetype)
+find_package(freetype CONFIG REQUIRED)
+
+hunter_add_package(SDL2)
+find_package(SDL2 CONFIG REQUIRED)
+
+set(SRC_C_FILES
+        SDL_ttf.c
+        )
+
+set(SRC_H_FILES
+        SDL_ttf.h
+        )
+
+set(SRC_FILES ${SRC_C_FILES} ${SRC_H_FILES})
+
+add_library(${PROJECT_NAME} ${SRC_FILES})
+
+target_link_libraries(${PROJECT_NAME} freetype::freetype SDL2::SDL2 SDL2::SDL2main)
+
+include(CMakePackageConfigHelpers)
+write_basic_package_version_file(
+        "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}ConfigVersion.cmake"
+        COMPATIBILITY SameMajorVersion
+)
+
+set(TARGETS_EXPORT_NAME "${PROJECT_NAME}Targets")
+configure_package_config_file(
+        "${CMAKE_CURRENT_LIST_DIR}/cmake/SDL_ttf_Config.cmake.in"
+        "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}Config.cmake"
+        INSTALL_DESTINATION "lib/cmake/${PROJECT_NAME}"
+)
+
+install(TARGETS ${PROJECT_NAME} EXPORT "${PROJECT_NAME}Targets"
+        LIBRARY DESTINATION "lib"
+        ARCHIVE DESTINATION "lib"
+        INCLUDES DESTINATION "include")
+
+install(
+        FILES
+        "SDL_ttf.h"
+        DESTINATION
+        "include/SDL2_ttf")
+
+install(
+        FILES
+        "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}Config.cmake"
+        "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}ConfigVersion.cmake"
+        "cmake/freetypeConfigMacro.cmake"
+        "cmake/SDL2ConfigMacro.cmake"
+        "cmake/FindSDL2.cmake"
```

Comment:
Not sure I understand the plans about
```
"cmake/freetypeConfigMacro.cmake"
"cmake/SDL2ConfigMacro.cmake"
"cmake/FindSDL2.cmake"
```
files, but I think that we should not touch them when `HUNTER_ENABLED=ON`.

Reply:
I think I need some guidance here, maybe there is something I do not understand correctly

If I create a new project and try to use SDL_ttf like this:

```
cmake_minimum_required(VERSION 3.0)
project(SDL_ttf_test)

find_package(SDL_ttf CONFIG REQUIRED)

add_executable(SDL_ttf_test main.cpp)
target_link_libraries(${PROJECT_NAME} SDL_ttf::SDL_ttf)
```
As I understand it, it would now try to link against `freetype::freetype`.
If I now use `find_package(Freetype MODULE REQUIRED)` from CMake, i would get the variables `FREETYPE_INCLUDE_DIRS` `FREETYPE_LIBRARIES`.
So, my understanding was that I create a Imported library `freetype::freetype` like this.

```
find_package(Freetype MODULE REQUIRED)

add_library(freetype::freetype UNKNOWN IMPORTED)
set_target_properties(
        freetype::freetype
        PROPERTIES
        INTERFACE_INCLUDE_DIRECTORIES "${FREETYPE_INCLUDE_DIRS}"
        IMPORTED_LOCATION "${FREETYPE_LIBRARIES}"
)
```

Then, If I want to build SDL_ttf, also without hunter, I have to to the same kind target definition withing "disabled-mode". Therefore I created this macros so I can just import in in both.

Is there a better way to do this ? Help is much appreciated !

Reply:
The logic is simple:

> "cmake/FindSDL2.cmake"

Such files should not be installed at all. They are shipped with projects locally:
* https://stackoverflow.com/a/20857070/2288008

> "cmake/SDL2ConfigMacro.cmake"

It doesn't sound like an `SDL_ttf` responsibility. From name of the file I expect it will be installed by `SDL2` project. Just imagine that there is `SDL_something_else` project that want to install `SDL2ConfigMacro` too - conflict.

> "cmake/freetypeConfigMacro.cmake"

Same here. I expect that anything that match `cmake/freetype*` installed by `freetype` project.

> Is there a better way to do this?

Use "disabled-mode" modules there too. They are designed to be shared accross the projects. That's why they added as submodule here: https://github.com/forexample/hunter-with-disabled-mode-example

It's not very nice, nice way to manage dependencies is to use Hunter :)